registry/storage/paths.go

```diff
@@ -367,6 +369,12 @@ type layerLinkPathSpec struct {
 
 func (layerLinkPathSpec) pathSpec() {}
 
+type layerDirectoryPathSpec struct {
```

Comment:
Please document this type. Notice the other comments around here and please try to follow that style.